File: 백준/Silver/1024. 수열의 합/수열의 합.py
N, L = map(int, input().split())

while 1:
    a = (1-L)/2 + N/L
    if a < 0 or L > 100:
        print(-1)
        break
    if a == int(a):
        for i in range(L):
            print(int(a + i), end=' ')
        break
    else:
        L += 1

File: 백준/Silver/1024. 수열의 합/수열의 합.js
const readline = require("readline")
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
})

rl.on("line", (line) => {
  const [N, L] = line.split(" ").map(Number)

  let _L = L

  while (true) {
    const a = (1 - _L) / 2 + N / _L

    if (a < 0 || _L > 100) {
      console.log(-1)
      break
    }

    if (a === Math.floor(a)) {
      const sequence = []
      for (let i = 0; i < _L; i++) {
        sequence.push(Math.floor(a + i))
      }
      console.log(sequence.join(" "))
      break
    } else {
      _L += 1
    }
  }

  rl.close()
})
